### PR TITLE
Add CA itemized deduction logic

### LINKED ISSUE #1904: Add formula for ca_itemized_deductions variable
The `ca_itemized_deductions.py` module does not have a formula.

### PR DESCRIPTION
**Pull request by martinholmer**
Fixes #1904 and adds `ca_income_tax` to the list of state income taxes.

Also, makes several parameter files and variable formulas more similar to each other; these changes improve code health but make no changes in logic.

And finally, improve the specification of several 2018 US integration tests so that they don't fail when adding `ca_income_tax` to the list of state income taxes.

### CODE REVIEW

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/itemized/ca_itemized_deductions.py [hunk 1]
```diff
@@ -12,3 +12,17 @@ class ca_itemized_deductions(Variable):
...
+            deduction
+            for deduction in p.itemized_deductions
+            if deduction not in ["salt_deduction"]
+        ]
+        itm_deds_less_salt = add(tax_unit, period, itm_deds)
+        property_taxes = add(tax_unit, period, ["real_estate_taxes"])
+        salt = p.itemized.salt_and_real_estate
+        cap = salt.cap[tax_unit("filing_status", period)]
+        capped_property_taxes = min_(property_taxes, cap)
+        return itm_deds_less_salt + capped_property_taxes
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ca/tax/income/deductions/itemized/ca_itemized_deductions.py:28]: I think California also caps itemized deductions based on AGI. We can leave that for another issue/PR though.

https://leginfo.legislature.ca.gov/faces/codes_displaySection.xhtml?lawCode=RTC&sectionNum=17077.
https://www.ftb.ca.gov/forms/2022/2022-540-ca-instructions.html (line 29 + itemized deduction worksheet)

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/itemized/ca_itemized_deductions.py [hunk 2]
```diff
@@ -12,3 +12,17 @@ class ca_itemized_deductions(Variable):
...
+        p = parameters(period).gov.irs.deductions
+        itm_deds = [
+            deduction
+            for deduction in p.itemized_deductions
+            if deduction not in ["salt_deduction"]
+        ]
+        itm_deds_less_salt = add(tax_unit, period, itm_deds)
+        property_taxes = add(tax_unit, period, ["real_estate_taxes"])
+        salt = p.itemized.salt_and_real_estate
+        cap = salt.cap[tax_unit("filing_status", period)]
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ca/tax/income/deductions/itemized/ca_itemized_deductions.py:26]: I don't think California caps SALT-less-state-income-tax.

From https://www.ftb.ca.gov/forms/2022/2022-540-ca.pdf
image

and https://www.ftb.ca.gov/forms/2022/2022-540-ca-instructions.html

>Line 5e – The federal deduction for state and local tax is limited to $10,000 ($5,000 for married filing separate) for the aggregate of state and local income taxes and property taxes. California does not conform. If your deduction was limited under federal law, enter an adjustment on line 5e, column C for the amount over the federal limit.

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/itemized/ca_itemized_deductions.py [hunk 3]
```diff
@@ -12,3 +12,17 @@ class ca_itemized_deductions(Variable):
         "https://www.ftb.ca.gov/forms/2022/2022-540.pdf"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ca/tax/income/deductions/itemized/ca_itemized_deductions.py:12]:
```suggestion
        "https://www.ftb.ca.gov/forms/2022/2022-540-ca-instructions.html"
```

**Review by MaxGhenis — COMMENTED** [review]
Thanks Martin, appreciate the cleanup.

**Comment by martinholmer** [comment]
@MaxGhenis,  Thanks for pointing out that I had missed the limitation of CA itemized deductions.   This pull request now includes that logic and associated parameters, and it also includes new unit tests.   So, I think this resolves all of your review issues.

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/itemized/ca_itemized_deductions.py (hunk 2 above)
Comment (martinholmer) [reply policyengine_us/variables/gov/states/ca/tax/income/deductions/itemized/ca_itemized_deductions.py]: Done

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/itemized/ca_itemized_deductions.py (hunk 1 above)
Comment (martinholmer) [reply policyengine_us/variables/gov/states/ca/tax/income/deductions/itemized/ca_itemized_deductions.py]: Done.

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/itemized/ca_itemized_deductions.py (hunk 3 above)
Comment (martinholmer) [reply policyengine_us/variables/gov/states/ca/tax/income/deductions/itemized/ca_itemized_deductions.py]: Done.

**Pull request merged**